pages/index.js

```diff
@@ -18,36 +21,129 @@ import {
   LatestBlockTXNIcon,
   SecondsToBlockIcon,
   TotalSupplyIcon,
-} from 'svgx'
+} from 'svgx';
 
+const LastBlockInfo = () => {
+  const cardWidth = useBreakpointValue({
+    base: '100%',
+    sm: 'calc(50% - 1rem)',
+    md: 'calc(33% - 1rem)',
+  });
+  const headBlock = useBlockHead();
 
-const useBlocks = () => ({
-  loaded: true,
-  data: new Array(14).fill({
-    sequence: 23940194,
-    size: 4096,
-    transactions_count: 1,
-    hash: '0000456456654564346345644564000045645665456434634564456445644564',
-    timestamp: '6/22/2021 8:09:45PM'
-  })
-})
+  const getValue = (field, transform = (value) => value) => {
+    return headBlock.loaded ? (
+      transform(headBlock.data[field])
+    ) : (
+      <span>&nbsp;</span>
+    );
+  };
+
+  return (
+    <Flex w="100%" wrap="wrap" mb="2.25rem">
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Difficulty"
+        value={getValue('difficulty')}
+        icon={<DifficultyIcon />}
+        isLoading={!headBlock.loaded}
+      />
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Height"
+        value={getValue('sequence')}
+        icon={<HeightIcon />}
+        isLoading={!headBlock.loaded}
+      />
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Latest block hash"
+        value={getValue('hash', truncateHash)}
+        icon={<LatestBlockHashIcon />}
+        isLoading={!headBlock.loaded}
+      />
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Latest Block txn"
+        value={getValue('transactions_count')}
+        icon={<LatestBlockTXNIcon />}
+        isLoading={!headBlock.loaded}
+      />
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Seconds to block"
+        value={getValue('time_since_last_block_ms', (value) =>
+          Math.floor(value / 1000),
+        )}
+        icon={<SecondsToBlockIcon />}
+        isLoading={!headBlock.loaded}
+      />
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Total Supply"
+        value="-"
+        icon={<TotalSupplyIcon />}
+        isLoading={!headBlock.loaded}
+      />
```

Comment:
Minor, but it might make sense to keep the values in an interable and then simply map over them?

```ts
const cards = [
  {
    label: 'Difficulty',
    value: getValue('difficulty'),
    icon: <DifficultyIcon />,
  },
  {
    label: 'Height',
    value: getValue('sequence'),
    icon: <HeightIcon />,
  },
  {
    label: 'Latest block hash',
    value: getValue('hash', truncateHash),
    icon: <LatestBlockHashIcon />,
  },
  {
    label: 'Latest Block txn',
    value: getValue('transactions_count'),
    icon: <LatstBlockTXNIcon />,
  },
  {
    label: 'Seconds to block',
    value: getValue('time_since_last_block_ms', value =>
      Math.floor(value / 1000)
    ),
    icon: <SecondsToBlockIcon />,
  },
  {
    label: 'Total Supply',
    value: '-',
    icon: <TotalSupplyIcon />,
  },
]

// [...]

cards.map(x => <Card
  key={x.label} 
  {...x}
  m="0.5rem"
  w={cardWidth}
  isLoading={!headBlock.loaded}
/>)

```

Reply:
I tryed to move cards into global variable. It's better then recreate this array at any rerender.